core/Tracker/VisitExcluded.php

```diff
@@ -89,8 +89,14 @@ public function isExcluded()
          * @param bool &$excluded Whether the request should be excluded or not. Initialized
          *                        to `false`. Event subscribers should set it to `true` in
          *                        order to exclude the request.
+         * @param Request $request The request object which contains all of the request's information
+         * @param bool|string $userAgent The user-agent used by the visitor
+         * @param bool|string $ip The IP-address of the visitor
+         *
+         * The additional parameters $userAgent and $ip are needed, because the values in the 
+         * object $request are empty when the constructor is called from the log-import.
```

Comment:
Sorry for being a bit annoying here. I understand they are empty when there is a log-import but I don't really understand why? Are they supposed to be ignored when doing a log-import? Do I need to treat them differently when they are set / not set? Is it ok to fallback to `$this->request`? Can I maybe instead always use the ones from `$this->request`? When IP and user agent is set, are they the same ones as in `$this->request`? :)